File: go/cars-assemble/cars_assemble.go
package cars

const CPC = 10_000
const CPT = 95_000

func CalculateWorkingCarsPerHour(productionRate int, successRate float64) float64 {
	return float64(productionRate) * successRate / 100
}

func CalculateWorkingCarsPerMinute(productionRate int, successRate float64) int {
	return int(CalculateWorkingCarsPerHour(productionRate, successRate) / 60)
}

func CalculateCost(count int) uint {
	return uint(count/10*CPT + count%10*CPC)
}

File: lua/cars-assemble/cars-assemble.lua
return {
    calculate_working_cars_per_hour = function(n, r) return n * r / 100 end,
    calculate_working_cars_per_minute = function(n, r) return (n * r / 100) // 60 end,
    calculate_cost = function(n) return 95000 * (n // 10) + 10000 * (n % 10) end
}
